### ISSUE TITLE
ProperlyCloseFilesTestCase.testProperlyCloseFiles written in an aggressively stupid manner

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2054]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2054|
|Type|defect|
|Created|2006-09-02 20:20:51Z|

Some problems:

 * Assumes the environment will not allow 2048 connections
 * Assumes the environment is capable of performing 2048 connections in 120 seconds (default trial timeout)
 * Has an utterly pointless assertion at the end of the run
 * Is factored poorly, given the code-sharing going on with test_ssl
 * Doesn't properly wait for connections to drop so as to avoid a warning from trial

<details><summary>Searchable metadata</summary>

```
trac-id__2054 2054
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__Core_2_5 Core-2.5
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1157228451000000 1157228451000000
changetime__1158813794000000 1158813794000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

Ready for review in properly-close-files-2054.

Whether the new approach is a sufficient improvement or not, I am uncertain.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @exarkun|
|-|-|

test_tcp conflicts when I try to unbranch. 

Can you please resolve the conflicts and assign back to me?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set owner to @exarkun|
|-|-|

The test definitely looks way more stable. 

Definitely fix these before merging:
 * Over 80 columns in:
   * test_ssl.py:179
   * test_ssl.py:184
   * test_tcp.py:776
   * test_tcp.py:826
   * test_tcp.py:834
 * Add a docstring to !StolenTCPTestCase

Think about these:
 * Conceivably, creating the ServerFactory could be moved out of createServer and into test_properlyCloseFiles. 
 * Ditto, mutatis mutandis, for connectClient
 * test_properlyCloseFiles is pretty huge and difficult to read. The docstrings help a lot, but it's still a pretty weird test. Make it simpler if possible. Maybe add comments to [#1945](https://github.com/twisted/twisted/issues/1945)

Assign back to me for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @jml|
|-|-|

Long lines shortened, createServer and connectClient factoring improved.

I moved the two class definitions out of test_properlyCloseFiles.  This makes the test method _almost_ fit on one page, and helps readability I think.  Not sure what else to do.  I think moving the remaining methods out would do as much to hamper readability as to help it.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set owner to @exarkun|
|-|-|

Beautiful. Merge away.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [18141]) Merge properly-close-files-2054-2

Author: exarkun
Reviewer: jml
Fixes [#2054](https://github.com/twisted/twisted/issues/2054)

Refactor the TCP and SSL tests which make sure platform socket resources are
cleaned up properly so that they do not need to open hundreds or thousands of
sockets and so that there is a clear boundary between the abstract
connection-oriented mixin and the concrete TCP and SSL implementations of it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `reopened`|
|-|-|

Commit reverted - the test fails on Windows.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Fixed on Windows, I think.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @jml<br>@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_12"></a>|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|

Merge away :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [18210]) Merge properly-close-files-2054-2

Author: exarkun
Reviewer: jml
Fixes [#2054](https://github.com/twisted/twisted/issues/2054)

Refactor the TCP and SSL tests which make sure platform socket resources are
cleaned up properly so that they do not need to open hundreds or thousands of
sockets and so that there is a clear boundary between the abstract
connection-oriented mixin and the concrete TCP and SSL implementations of it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
